hal/src/stm32f2xx/i2c_hal.c

```diff
@@ -906,10 +906,7 @@ void HAL_I2C_Flush_Data(HAL_I2C_Interface i2c, void* reserved)
 
 bool HAL_I2C_Is_Enabled(HAL_I2C_Interface i2c, void* reserved)
 {
-    HAL_I2C_Acquire(i2c, NULL);
-    bool en = i2cMap[i2c]->I2C_Enabled;
-    HAL_I2C_Release(i2c, NULL);
-    return en;
+    return i2cMap[i2c]->I2C_Enabled;
```

Comment:
volatile.

Reply:
USART serial `HAL_USART_Is_Enabled` should also be using a volatile variable underneath.

Reply:
Same to `HAL_USB_USART_Is_Enabled()`?

Reply:
Probably not. We only care about methods used in `pinAvailable()`.